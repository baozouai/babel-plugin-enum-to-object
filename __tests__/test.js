/* global it, expect */

import { transformAsync } from '@babel/core'
import plugin from '../dist/babel-plugin-enum-to-object'

const defaultOptions = {
  plugins: [plugin],
}

const noReflectOptions = {
  plugins: [[plugin, { reflect: false }]],
}
const numberEnumInput = `export enum Direction {
  Up,
  Down,
  Left,
  Right,
}  
`
const stringEnumInput = `export enum Direction {
  Up = 'Up',
  Down = 'Down',
  Left = 'Left',
  Right = 'Right',
}`

const mixEnumInput = `export enum Status {
  PAID,
  UNPAID = '21',
  PART = 44,
  HALF
}
`

it('Transforms NumericLiteral enum', async () => {
  const { code: output } = await transformAsync(numberEnumInput, defaultOptions)
  expect(output).toMatchSnapshot()
})

it('Transforms string enum', async () => {
  const { code: output } = await transformAsync(stringEnumInput, defaultOptions)
  expect(output).toMatchSnapshot()
})

it('Transforms string number mix', async () => {
  const { code: output } = await transformAsync(mixEnumInput, defaultOptions)
  expect(output).toMatchSnapshot()
})

it('Transforms NumericLiteral enum  with reflect false', async () => {
  const { code: output } = await transformAsync(numberEnumInput, noReflectOptions)
  expect(output).toMatchSnapshot()
})

it('Transforms string enum  with reflect false', async () => {
  const { code: output } = await transformAsync(stringEnumInput, noReflectOptions)
  expect(output).toMatchSnapshot()
})

it('Transforms string number mix  with reflect false', async () => {
  const { code: output } = await transformAsync(mixEnumInput, noReflectOptions)
  expect(output).toMatchSnapshot()
})
