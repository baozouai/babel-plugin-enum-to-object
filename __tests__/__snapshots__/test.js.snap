// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transforms NumericLiteral enum  with reflect false 1`] = `
"export const Direction = {
  Up: 0,
  Down: 1,
  Left: 2,
  Right: 3
};"
`;

exports[`Transforms NumericLiteral enum 1`] = `
"export const Direction = {
  Up: 0,
  0: "Up",
  Down: 1,
  1: "Down",
  Left: 2,
  2: "Left",
  Right: 3,
  3: "Right"
};"
`;

exports[`Transforms string enum  with reflect false 1`] = `
"export const Direction = {
  Up: 'Up',
  Down: 'Down',
  Left: 'Left',
  Right: 'Right'
};"
`;

exports[`Transforms string enum 1`] = `
"export const Direction = {
  Up: 'Up',
  Down: 'Down',
  Left: 'Left',
  Right: 'Right'
};"
`;

exports[`Transforms string number mix  with reflect false 1`] = `
"export const Status = {
  PAID: 0,
  UNPAID: '21',
  PART: 44,
  HALF: 45
};"
`;

exports[`Transforms string number mix 1`] = `
"export const Status = {
  PAID: 0,
  0: "PAID",
  UNPAID: '21',
  21: "UNPAID",
  PART: 44,
  44: "PART",
  HALF: 45,
  45: "HALF"
};"
`;
